examples/tutorial.py

```diff
@@ -929,13 +1038,402 @@ def get_overlapping_labels(array1, array2):
+#     ```
+#
+
+# %% [markdown]
+# ## Daisy Tasks
+
+# %% [markdown]
+# ### A Simple Task
+
 # %%
+import daisy  # noqa
+from daisy import Coordinate, Roi  # noqa
+import time  # noqa
+
+# Create a super simple task
+dummy_task = daisy.Task(
+    "Dummy",  # We give the task a name
+    total_roi=Roi((0,), (100,)),  # a 1-D bounding box [0,100)
+    read_roi=Roi((0,), (1,)),  # We read in blocks of size [0,1)
+    write_roi=Roi((0,), (1,)),  # We write in blocks of size [0, 1)
+    process_function=lambda block: time.sleep(
+        0.05
+    ),  # Our process function takes the block and simply waits
+    num_workers=5,
+)
+
+# execute the task without any multiprocessing
+daisy.run_blockwise([dummy_task], multiprocessing=False)
+daisy.run_blockwise([dummy_task], multiprocessing=True)
+
+# %% [markdown]
+# Since there are 100 blocks to process, and we always sleep for 0.05 seconds, it takes 5 seconds
+# to run all the blocks without multiprocessing, but only 1 second with `multiprocessing=True`.
+# When you run a task with `daisy`, you get a progress bar along with a summary
+# of the execution after you have finished processing. Here no errors were thrown so we completed all 100 blocks.
+#
+# `daisy` also programatically returns a dictionary containing a summary of the execution for all given
+# tasks so that you can decide how to procede based on whether any blocks failed.
+#
+# Next lets look see what we can do to modify this task
+
+# %% [markdown]
+# ### Failing blocks
+# %%
+
+import random  # noqa
+
+
+def random_fail(block: daisy.Block):
+    if random.random() < 0.2:
+        block.status = daisy.BlockStatus.FAILED
+    return block
+
+
+# Create a super simple task
+failing_task = daisy.Task(
+    "Failures",  # We give the task a name
+    total_roi=Roi((0,), (100,)),  # a 1-D bounding box [0,100)
+    read_roi=Roi((0,), (1,)),  # We read in blocks of size [0,1)
+    write_roi=Roi((0,), (1,)),  # We write in blocks of size [0, 1)
+    process_function=random_fail,  # Our process function takes the block and simply waits
+    max_retries=0,  # We do not retry failed blocks to get expected failure counts
+)
+
+# execute the task without any multiprocessing
+daisy.run_blockwise([failing_task], multiprocessing=False)
+
+# %% [markdown]
+
+# In this case, we have a 20% chance of failing each block. As you see from the execution summary,
+# about 20% of the blocks failed.
+# If your blocks are failing, you may want to rerun the task and only process failed blocks. This
+# is where we use the `check_function` argument in the `daisy.Task`.
+
+# %%
+import tempfile  # noqa
+from pathlib import Path  # noqa
+
+with tempfile.TemporaryDirectory() as tmpdir:
+
+    def random_fail(block: daisy.Block):
+        if random.random() < 0.2:
+            block.status = daisy.BlockStatus.FAILED
+        else:
+            Path(tmpdir, f"{block.block_id[1]}").touch()
+        return block
+
+    def check_block(block: daisy.Block) -> bool:
```

Comment:
If the quickstart is first, you can link to the in-depth tutorial locations. E.g. "For more information about the performance implications of the check function, look here (link to in depth section)"